### PR TITLE
Dist

### PR DESCRIPTION
**Pull request by shreyashekhar**
- Implemented Google Maps Distance Matrix API to find distances between lat/long coordinates and addresses
- Implemented functionality to find current location of the individual from phone in lat/long coordinates
- Can find distance between phone location and an event location
- Adds events to adapter for timeline based on distance < 60 miles from individual
- On profile page, displays all events created by individual (regardless of distance) , and only displays events once (no endless scroll on profile page) to avoid confusion
- added fixes for api

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,14 +1 @@
-*.iml
-.gradle
-/local.properties
-/.idea/caches
-/.idea/libraries
-/.idea/modules.xml
-/.idea/workspace.xml
-/.idea/navEditor.xml
```
Comment (rfaisal) [inline .gitignore:8]: why are these removed?

File: app/src/main/java/com/example/solidarity/ProfileFragment.java [hunk 2]
```diff
@@ -68,9 +75,8 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         FrameLayout flayout = view.findViewById(R.id.fLayout);
         profImage = view.findViewById(R.id.ivProfileImage);
         swipeContainer = (SwipeRefreshLayout) view.findViewById(R.id.swipeContainer);
-        /**profImage = new CircleImageView(getContext());
```
Comment (rfaisal) [inline app/src/main/java/com/example/solidarity/ProfileFragment.java:71]: no dead code

File: app/src/main/java/com/example/solidarity/fragments/EventsFragment.java [hunk 3]
```diff
@@ -121,12 +167,53 @@ public void done(List<Event> events, ParseException e) {
...
                 swipeContainer.setRefreshing(false);
-
-
             }
         });
+    }
 
+    public void getDist(String fromLat, String fromLong, String toAddress,
+                        final ArrayList<Integer> distances, final Event event) {
+        String URL_4 = "https://maps.googleapis.com/maps/api/distancematrix/json?uni" +
```
Comment (rfaisal) [inline app/src/main/java/com/example/solidarity/fragments/EventsFragment.java:180]: should you be using String.format?

Comment (rfaisal) [inline app/src/main/java/com/example/solidarity/fragments/EventsFragment.java:180]: Why is the variable named URL_4 ? what is 4 and why is it all caps?

File: app/src/main/java/com/example/solidarity/fragments/EventsFragment.java [hunk 4]
```diff
@@ -121,12 +167,53 @@ public void done(List<Event> events, ParseException e) {
...
+                "ts=imperial&origins=" + fromLat + "," + fromLong + "&destinations=" +
+                toAddress + "&key=" + API_KEY;
+        client.get(URL_4, new JsonHttpResponseHandler() {
+            @Override
+            public void onSuccess(int statusCode, Headers headers, JsonHttpResponseHandler.JSON json) {
+                JSONObject jsonObject = json.jsonObject;
+                try {
+                    JSONArray dist = (JSONArray) jsonObject.get("rows");
+                    JSONObject obj2 = (JSONObject) dist.get(0);
```
Comment (rfaisal) [inline app/src/main/java/com/example/solidarity/fragments/EventsFragment.java:189]: obj2 is not  good naming

File: app/src/main/java/com/example/solidarity/fragments/EventsFragment.java [hunk 5]
```diff
@@ -121,12 +167,53 @@ public void done(List<Event> events, ParseException e) {
...
+                    JSONObject obj3 = (JSONObject) disting.get(0);
+                    JSONObject obj4 = (JSONObject) obj3.get("distance");
+                    String distance = (String) obj4.get("text");
+                    if (distance.contains(".")) {
+                        distance = distance.substring(0, distance.indexOf("."));
+                    }
+                    distance = distance.replaceAll(",", "");
+                    int dis = 0;
+                    if (distance.contains(" ")) {
+                        dis = Integer.valueOf(distance.substring(0, distance.indexOf(" ")));
```
Comment (rfaisal) [inline app/src/main/java/com/example/solidarity/fragments/EventsFragment.java:200]: can these throw errors? should you use a default ?

File: app/src/main/java/com/example/solidarity/fragments/EventsFragment.java [hunk 6]
```diff
@@ -121,12 +167,53 @@ public void done(List<Event> events, ParseException e) {
...
 
+    public void getDist(String fromLat, String fromLong, String toAddress,
+                        final ArrayList<Integer> distances, final Event event) {
+        String URL_4 = "https://maps.googleapis.com/maps/api/distancematrix/json?uni" +
+                "ts=imperial&origins=" + fromLat + "," + fromLong + "&destinations=" +
+                toAddress + "&key=" + API_KEY;
+        client.get(URL_4, new JsonHttpResponseHandler() {
+            @Override
+            public void onSuccess(int statusCode, Headers headers, JsonHttpResponseHandler.JSON json) {
+                JSONObject jsonObject = json.jsonObject;
```
Comment (rfaisal) [inline app/src/main/java/com/example/solidarity/fragments/EventsFragment.java:186]: This function is not well done in terms of code quality. Follow the example you learned from your assignment and create a class and construct the class from this JSONObject

**Review by rfaisal — CHANGES_REQUESTED** [review]

**Comment by rfaisal** [comment]
can you name the pull request better?

File: .gitignore (hunk 1 above)
Comment (shreyashekhar) [reply .gitignore:8]: I think these were removed by accident when I added secrets.xml to the file- will fix this!

File: app/src/main/java/com/example/solidarity/fragments/EventsFragment.java (hunk 3 above)
Comment (shreyashekhar) [reply app/src/main/java/com/example/solidarity/fragments/EventsFragment.java:180]: Just incorporated String.format instead

Comment (shreyashekhar) [reply app/src/main/java/com/example/solidarity/fragments/EventsFragment.java:180]: I was testing out different URL formats, and forgot to rename this- sorry about that!

File: app/src/main/java/com/example/solidarity/ProfileFragment.java (hunk 2 above)
Comment (shreyashekhar) [reply app/src/main/java/com/example/solidarity/ProfileFragment.java:71]: I had just removed this unused code in this PR- all irrelevant comments should be gone
